src/components/play-audio/play-audio.js

```diff
@@ -0,0 +1,128 @@
+import React, { useState, useEffect, createRef } from 'react';
+import PlayIcon from '-!svg-react-loader!../../images/SSWPlay2x.svg';
+import PauseIcon from '-!svg-react-loader!../../images/SSWPause2x.svg';
+import lottie from 'lottie-web';
+import animation from '../../animations/lf30_editor_DirdRw.json';
+import PropTypes from 'prop-types';
+
+const PlayAudio = ({ hasAnimation, audioSrc }) => {
+  const animationContainer = createRef();
+  const [audio, setAudio] = useState({});
+  const [isPlaying, setIsPlaying] = useState(false);
+  const [hoverAudio, setHoverAudio] = useState(false);
+
+  useEffect(() => {
+    setAudio(new Audio());
+    if (hasAnimation) {
+      lottie.loadAnimation({
+        container: animationContainer.current,
+        render: 'svg',
+        loop: true,
+        animationData: animation,
+      });
+      lottie.stop();
+      return () => lottie.destroy();
+    }
+  }, []);
+
+  const playAudio = () => {
+    setIsPlaying(true);
+    audio.src = audioSrc;
+    audio.play();
+    if (hasAnimation) {
+      lottie.play();
+    }
+  };
+  const stopAudio = () => {
+    setIsPlaying(false);
+    audio.pause();
+    audio.currentTime = 0;
+    if (hasAnimation) {
+      lottie.stop();
+    }
+  };
+
+  audio.onended = () => {
+    setIsPlaying(false);
+    if (hasAnimation) {
+      lottie.stop();
+    }
+  };
+
+  return (
+    <>
+      {hasAnimation ? (
+        <div className={'pt-1 pl-2 flex'}>
+          {!isPlaying && (
+            <PlayIcon
+              aria-label="play audio"
+              className={'cursor-pointer mr-2'}
+              onClick={() => {
+                playAudio();
+              }}
+            />
+          )}
+          {isPlaying && (
+            <PauseIcon
+              aria-label="pause audio"
+              className={'cursor-pointer mr-2'}
+              onClick={() => {
+                stopAudio();
```

Comment:
Does this pause actually stop the audio and revert the current time back to 0?

Reply:
You are right, I shouldn't be reverting the audio current time back to 0.... I changed the function name and it just pauses the audio.